blocks/library/embed/index.js

```diff
@@ -34,188 +34,226 @@ function toggleAlignment( align ) {
-			this.unmounting = true;
-		}
+			componentWillMount() {
+				if ( this.props.attributes.url ) {
+					// if the url is already there, we're loading a saved block, so we need to render
+					// a different thing, which is why this doesn't use 'fetching', as that
+					// is for when the user is putting in a new url on the placeholder form
+					this.setState( { fetching: true } );
+					this.doServerSideRender();
+				}
+			}
 
-		doServerSideRender( event ) {
-			if ( event ) {
-				event.preventDefault();
+			componentWillUnmount() {
+				// can't abort the fetch promise, so let it know we will unmount
+				this.unmounting = true;
 			}
-			const { url } = this.props.attributes;
-			const api_url = wpApiSettings.root + 'oembed/1.0/proxy?url=' + encodeURIComponent( url ) + '&_wpnonce=' + wpApiSettings.nonce;
-
-			this.setState( { error: false, fetching: true } );
-			window.fetch( api_url, {
-				credentials: 'include',
-			} ).then(
-				( response ) => {
-					if ( this.unmounting ) {
-						return;
-					}
-					response.json().then( ( obj ) => {
-						const { html, type } = obj;
-						if ( html ) {
-							this.setState( { html, type } );
-						} else {
-							this.setState( { error: true } );
-						}
-						this.setState( { fetching: false } );
-					} );
-				}
-			);
-		}
 
-		render() {
-			const { html, type, error, fetching } = this.state;
-			const { url, caption } = this.props.attributes;
-			const { setAttributes, focus, setFocus } = this.props;
+			doServerSideRender( event ) {
+				if ( event ) {
+					event.preventDefault();
+				}
+				const { url } = this.props.attributes;
+				const api_url = wpApiSettings.root + 'oembed/1.0/proxy?url=' + encodeURIComponent( url ) + '&_wpnonce=' + wpApiSettings.nonce;
 
-			if ( fetching ) {
-				return (
-					<div className="blocks-embed is-loading">
-						<Spinner />
-						<p>{ wp.i18n.__( 'Embedding…' ) }</p>
-					</div>
+				this.setState( { error: false, fetching: true } );
+				window.fetch( api_url, {
+					credentials: 'include',
+				} ).then(
+					( response ) => {
+						if ( this.unmounting ) {
+							return;
+						}
+						response.json().then( ( obj ) => {
+							const { html, type } = obj;
+							if ( html ) {
+								this.setState( { html, type } );
+							} else {
+								this.setState( { error: true } );
+							}
+							this.setState( { fetching: false } );
+						} );
+					}
 				);
 			}
 
-			if ( ! html ) {
+			render() {
+				const { html, type, error, fetching } = this.state;
+				const { url, caption } = this.props.attributes;
+				const { setAttributes, focus, setFocus } = this.props;
+
+				if ( fetching ) {
+					return (
+						<div className="blocks-embed is-loading">
+							<Spinner />
+							<p>{ wp.i18n.__( 'Embedding…' ) }</p>
+						</div>
+					);
+				}
+
+				if ( ! html ) {
+					return (
+						<Placeholder icon={ icon } label={ wp.i18n.__( title + ' URL' ) } className="blocks-embed">
```

Comment:
We cannot statically analyze dynamic concatenation like this when extracting strings. The arguments to translation functions must be strings. We'll likely want to consider an ESLint rule like one we have for Calypso, where the same applies:

https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/docs/rules/i18n-no-variables.md

Instead, we should use [`sprintf`](https://github.com/WordPress/gutenberg/tree/master/i18n#api) placeholders:

```js
sprintf( __( '%s URL' ), title )
```

https://github.com/alexei/sprintf.js#format-specification

Sometimes placeholders can become an issue for some languages which may vary the translation by gender, but in this case since they're product names I don't think they'd ever become gender-ified.